app/main.py

```diff
@@ -1,3 +1,60 @@
+from __future__ import annotations
+
+
 class Distance:
     # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += + other
+        return self
```

Comment:
Add must not change self. It must create new Distance obj

Reply:
def __iadd__(self, other: Distance | int | float) -> Distance:
        if isinstance(other, (int, float)):
            distance_new = Distance(self.km + other)
        else:
            distance_new = Distance(self.km + other.km)
        return distance_new

Like this?